src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateClone, obj.extraData);
+      stateHistory.push({ ...stateClone });
```

Comment:
It is better to push a copy of the object to the array at the end of each loop cycle, but outside of the `switch` block, after the current action is processed. This will ensure that the state history is updated consistently after each action.